nautobot/circuits/views.py

```diff
@@ -4,277 +4,189 @@
-        RequestConfig(request, paginate).configure(circuits_table)
-
-        return {
-            "circuits_table": circuits_table,
-        }
-
-
-class CircuitTypeEditView(generic.ObjectEditView):
-    queryset = CircuitType.objects.all()
-    model_form = forms.CircuitTypeForm
-
-
-class CircuitTypeDeleteView(generic.ObjectDeleteView):
-    queryset = CircuitType.objects.all()
-
-
-class CircuitTypeBulkImportView(generic.BulkImportView):
-    queryset = CircuitType.objects.all()
-    model_form = forms.CircuitTypeCSVForm
-    table = tables.CircuitTypeTable
+        context = super().get_extra_context(request, instance)
+        if self.action == "retrieve":
+            circuits = (
+                Circuit.objects.restrict(request.user, "view")
+                .filter(type=instance)
+                .prefetch_related("type", "tenant", "terminations__site")
+            )
 
+            circuits_table = tables.CircuitTable(circuits)
+            circuits_table.columns.hide("type")
 
-class CircuitTypeBulkDeleteView(generic.BulkDeleteView):
-    queryset = CircuitType.objects.annotate(circuit_count=count_related(Circuit, "type"))
-    table = tables.CircuitTypeTable
+            paginate = {
+                "paginator_class": EnhancedPaginator,
+                "per_page": get_paginate_count(request),
+            }
+            RequestConfig(request, paginate).configure(circuits_table)
+            context["circuits_table"] = circuits_table
+        return context
 
 
-#
-# Circuits
-#
+class CircuitTerminationUIViewSet(NautobotUIViewSet):
+    model = CircuitTermination
+    form_class = forms.CircuitTerminationForm
+    lookup_field = "pk"
+    queryset = CircuitTermination.objects.all()
+    serializer_class = serializers.CircuitTerminationSerializer
 
+    def get_object(self):
+        obj = super().get_object()
+        if self.action in ["create", "update"] and "circuit" in self.kwargs:
+            obj.circuit = get_object_or_404(Circuit, pk=self.kwargs["circuit"])
+        return obj
 
-class CircuitListView(generic.ObjectListView):
-    queryset = Circuit.objects.prefetch_related("provider", "type", "tenant", "termination_a", "termination_z")
-    filterset = filters.CircuitFilterSet
-    filterset_form = forms.CircuitFilterForm
-    table = tables.CircuitTable
+    def get_return_url(self, request, obj):
+        return obj.circuit.get_absolute_url()
 
 
-class CircuitView(generic.ObjectView):
-    queryset = Circuit.objects.all()
+class ProviderUIViewSet(NautobotUIViewSet):
+    model = Provider
+    bulk_create_form_class = forms.ProviderCSVForm
+    bulk_update_form_class = forms.ProviderBulkEditForm
+    filterset_class = filters.ProviderFilterSet
+    filterset_form_class = forms.ProviderFilterForm
+    form_class = forms.ProviderForm
+    queryset = Provider.objects.annotate(count_circuits=count_related(Circuit, "provider"))
+    serializer_class = serializers.ProviderSerializer
+    table_class = tables.ProviderTable
```

Comment:
As we improve the "Model Rules Everything" design, we can strip even more of this out to have these forms and classes be derived from the model itself. Think `model.form_class`, `model.serializer_class`, etc.